fdbserver/DDTeamCollection.actor.cpp

```diff
@@ -2917,7 +2917,25 @@ class DDTeamCollectionImpl {
 		if (res.size() > 0) {
 			// SOMEDAY: support wiggle multiple SS at once
 			ASSERT(!self->wigglingId.present()); // only single process wiggle is allowed
-			self->wigglingId = res.begin()->first;
+
+			state Optional<Value> localityKey;
+			state Optional<Value> localityValue;
+			if (self->configuration.perpetualStorageWiggleLocality != "0") {
+				ParsePerpetualStorageWiggleLocality(
+				    self->configuration.perpetualStorageWiggleLocality, &localityKey, &localityValue);
+			}
+
+			// if perpetual_storage_wiggle_locality has value and not 0(disabled).
+			if (localityKey.present()) {
+				auto server = self->server_info.at(res.begin()->first);
```

Comment:
The UID is not guaranteed to be valid here. Because it's possible the SS with this ID is already removed. Perpetual wiggler handles this situation in the function `excludeStorageServersForWiggle`, but we can do check in the above if block

Reply:
Done.